Case study + images (#217)
NoroffFEU/development-platforms: src/content/docs/devtools/Docker/docker-swarm-case-study.md

> | Attribute  | Details                                                      |

Reviewer: Please double check the formatting for markdown `frontmatter`. Here is an example:

```md
---
title: TensorFlow
author: Vegard Lekve <vegardlekve>
tags: Machine learning, Neural networks, JavaScript, Python
---
```

|------------|--------------------------------------------------------------|
| Title      | Docker swarm case study                                            |
| Author     | Malene Ivy Olsen                                   |
| Tags       | docker, container, containerization, swarm, container orchestration, kubernetes | <br/>
<br/>

# Docker Swarm - container orchestration tool
[Acronyms used in the text](#acronyms-used)
## Introduction

> This case study aims to compare Docker and Kubernetes, the two key technologies in containerization and orchestration. I'll begin by introducing Docker, an essential tool for containerization, to set the groundwork for understanding container orchestration tools. <br/>

Reviewer: ```suggestion
This case study aims to compare Docker and Kubernetes, the two key technologies in containerization and orchestration. I'll begin by introducing Docker, an essential tool for containerization, to set the groundwork for understanding container orchestration tools.

```

Reviewer: We do not use <br> tags in markdown files. A line break will achieve the same thing. Please remove all instances of <br>

> Then I'll dive into comparison of Docker Swarm and Kubernetes to help students/beginners understand these important tools in modern software development. 

Reviewer: ```suggestion
Then we will dive into comparison of Docker Swarm and Kubernetes to help students/beginners understand these important tools in modern software development. 
```

We do not use the first person in technical literature. If you need to refer to yourself you can as `The Author`, for example:

> In the Author's opinion...

## Containerization
A container works as a self-contained environment that holds all of the necessary components for running an application. It provides an isolated environment for running applications and services. <br/>
Containerization enables teams to package applications, dependencies and configurations into container images. This simplifies the development of microservices, which are small, independent software components that focus on specific tasks within a larger application. <br/>

### Container Key Milestones

> ![Container key milestones after 2013](../../../../assets/devtools/docker/docker-history-timeline.jpg)

Reviewer: @jaaneh Is there a better way to write these paths?

Reviewer 2: You can use `~/assets/*`

> In 1979, the concept of containers first emerged. This was 34 years before Docker surfaced in 2013. <br/>

Reviewer: There are 43 instances of <br> to be removed, I will no longer be commenting on them.

People often mix up images and containers, but they're actually two different parts of Docker. To sum it up; the image is like the package, and the container is what runs when you open the package. A single image can create multiple containers.

### Dockerfile and image layers
To build your own image, you'll need to compose a Dockerfile, creating a straightforward syntax with steps that is needed for its creation and execution. Each instruction in the Dockerfile contributes to a layer within the image. Each layer represents a set of filesystem changes, such as adding, modifying, or deleting files. Every time the Dockerfile changes, only the layers which have changes, are rebuilt. 

Some common Dockerfile instructions:
* **FROM**: Specifies the base image to use for the new image. 
* **COPY**: Copies files/directories from the host machine into the image.
* **RUN**: Executes commands withing the image.
* **EXPOSE**: Container will listen on port 3000 on runtime.
* **CMD** Specifies the default command to run when a container is started from the image. 

### Demonstration: build an image and run container

> I'll be demonstrating Dockers [welcome-to-docker](https://github.com/docker/welcome-to-docker) repo on Github. 

Reviewer: ```suggestion
Let's take a look at Dockers [welcome-to-docker](https://github.com/docker/welcome-to-docker) repo on Github. 
```

> 1. Install Docker Desktop on [Mac](https://docs.docker.com/desktop/install/mac-install/), [Windows](https://docs.docker.com/desktop/install/windows-install/) or [Linus](https://docs.docker.com/desktop/install/linux-install/). <br/> I am using a Mac, so this demonstration is for Mac users. 

Reviewer: ```suggestion
1. Install Docker Desktop on [Mac](https://docs.docker.com/desktop/install/mac-install/), [Windows](https://docs.docker.com/desktop/install/windows-install/) or [Linux](https://docs.docker.com/desktop/install/linux-install/).  We will continue with the Mac installation instructions. 
```

2. Make sure to make a folder where you can try this out. Open terminal and navigate to your folder before you clone the repo with this command: <br/>

>    ```

Reviewer: ```suggestion
   ```sh
```

Reviewer: Including the language adds highlighting
